airbyte-cdk/bulk/toolkits/load-avro/src/main/kotlin/io/airbyte/cdk/load/data/avro/AirbyteValueToAvroRecord.kt

```diff
@@ -4,73 +4,184 @@
-                airbyteValue.values.forEach { (name, value) ->
-                    recordSchema.getField(name)?.let { field ->
-                        record.put(name, convert(value, field.schema()))
+    //    fun convert(airbyteValue: AirbyteValue, schema: Schema): Any? {
+    //        when (airbyteValue) {
+    //            is ObjectValue -> {
+    //                val recordSchema =
+    //                    if (schema.type == Schema.Type.UNION) {
+    //                        schema.types.find { it.type == Schema.Type.RECORD }
+    //                            ?: throw IllegalArgumentException("Union must contain a record
+    // type")
+    //                    } else {
+    //                        schema
+    //                    }
+    //                val record = GenericData.Record(recordSchema)
+    //                airbyteValue.values.forEach { (name, value) ->
+    //                    val nameMangled = Transformations.toAvroSafeName(name)
+    //                    recordSchema.getField(nameMangled)?.let { field ->
+    //                        record.put(nameMangled, convert(value, field.schema()))
+    //                    }
+    //                }
+    //                return record
+    //            }
+    //            is ArrayValue -> {
+    //                val arraySchema =
+    //                    if (schema.type == Schema.Type.UNION) {
+    //                        schema.types.find { it.type == Schema.Type.ARRAY }
+    //                            ?: throw IllegalArgumentException("Union must contain an array
+    // type")
+    //                    } else {
+    //                        schema
+    //                    }
+    //                val array = GenericData.Array<Any>(airbyteValue.values.size, arraySchema)
+    //                airbyteValue.values.forEach { value ->
+    //                    array.add(convert(value, arraySchema.elementType))
+    //                }
+    //                return array
+    //            }
+    //            is BooleanValue -> return airbyteValue.value
+    //            is DateValue ->
+    //                throw IllegalArgumentException("String-based date types are not supported")
+    //            is IntegerValue -> return airbyteValue.value.toLong()
+    //            is IntValue -> return airbyteValue.value
+    //            is NullValue -> return null
+    //            is NumberValue -> return airbyteValue.value.toDouble()
+    //            is StringValue -> return airbyteValue.value
+    //            is TimeValue ->
+    //                throw IllegalArgumentException("String-based time types are not supported")
+    //            is TimestampValue ->
+    //                throw IllegalArgumentException("String-based timestamp types are not
+    // supported")
+    //            is UnknownValue -> throw IllegalArgumentException("Unknown type is not supported")
+    //        }
+    //    }
```

Comment:
Do we want to leave this block of commented out code?

Reply:
It's removed in the last commit